datumaro/components/converter.py

```diff
@@ -34,8 +36,22 @@ def convert(cls, extractor, save_dir, **options):
         return converter.apply()
 
     @classmethod
+    @error_rollback
     def patch(cls, dataset, patch, save_dir, **options):
-        return cls.convert(dataset, save_dir, **options)
+        # Is not any better in performance than just writing a dataset,
+        # but in case of patching (i.e. writing to the previous location),
+        # it allows to avoid many problems with removing and replacing
+        # existing files. Surely, this approach also has problems with
+        # removal of the given directory.
```

Comment:
I think this should be elaborated on. What are the problems with removal of the given directory?

Reply:
I see problems can occur if we can't remove a directory, or want to reuse the given directory. For example, if it is mounted or (sym-)linked. Ideally, I think, it would be better to wipe directory contents and write new data there.

Reply:
Okay, so reflect that in your comment.